### ISSUE TITLE
Update NixOS docs to reference the `conduwuit` module and package available in `nixos-unstable`? And include a migration guide?

### ISSUE THREAD

**Issue by sodiboo** [girlbossceo/conduwuit#650]
`nixos-unstable` now has a `services.conduwuit` module.

- https://github.com/NixOS/nixpkgs/pull/353651

(yes, i know you were involved in that discussion, and this isn't *news*. but the docs are outdated and didn't contain a solution when my homeserver unexpectedly broke. woe is me)

the documentation for conduwuit suggests there exists no such module, and you should use the `matrix-conduit` module instead.

https://github.com/girlbossceo/conduwuit/blob/c53e9e07f23756d92ecd124da10a9a970756977c/docs/deploying/nixos.md?plain=1#L42-L88

that's all fine, and has been working for a while for me, but the outdated advice there is also just **wrong** now, as it leads to the following error:

![conduit.service unable to start because it can't find conduwuit/bin/conduit. the actual binary is named conduwuit](https://github.com/user-attachments/assets/e22455c8-2b2f-426f-9a32-a7c00496bc17)

(yes, this is a somewhat trivial fix, but it's not like this is gonna be the last incompatibility between diverging modules. the conduwuit module has no such issue when running conduwuit)

the sections of documentation about sqlite and unix sockets can just be deleted outright, as these are workarounds to issues solved by introducing the new module (and that's the whole motivation for introducing a new module, anyhow)

---

tangentially related: i also think that sentence ought to say "available in nixpkgs" somewhere in there:

https://github.com/girlbossceo/conduwuit/blob/c53e9e07f23756d92ecd124da10a9a970756977c/docs/deploying/nixos.md?plain=1#L9

i would open a PR, but that single line change seems not important enough in isolation, and i don't have the energy (or really the knowledge necessary) to rewrite the documentation on how best to deploy it on nixos

**Comment by sodiboo** [comment]
It would also be REALLY NICE to have a neat migration guide, because i notice that e.g. the database path looks very suspicious:

- [`services.matrix-conduit` wants `/var/lib/matrix-conduit/`](https://github.com/NixOS/nixpkgs/blob/634fd46801442d760e09493a794c4f15db2d0cbb/nixos/modules/services/matrix/conduit.nix#L77)
- [`services.conduit` wants `/var/lib/conduwuit`](https://github.com/NixOS/nixpkgs/blob/634fd46801442d760e09493a794c4f15db2d0cbb/nixos/modules/services/matrix/conduwuit.nix#L144)

So just swapping out which module i use looks like it won't properly use my old database, at *least*. I could probably figure this out in an afternoon, but it feels like there ought to be a single resource that everyone can reference so it won't take them an entire afternoon.

**Comment by girlbossceo** [comment]
I've read this issue multiple times and I'm not sure what you're really asking from me. It's difficult to read it because of the very hostile tone this was written like and I'm inclined to just close it and re-open it written less hostile and more clear.

**Comment by GearKite** [comment]
Since there is a [`services.conduwuit`](https://search.nixos.org/options?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=conduwuit) module in nixpkgs now, it's probably preferred to use that over [`services.matrix-conduit`](https://search.nixos.org/options?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=matrix-conduit) (the docs are outdated).

I might be able to make a pull request for this.

**Comment by girlbossceo** [comment]
Yes if the conduwuit module has finally landed into nixpkgs general availability then that is preferred. It does require one change though when 0.5.0 is released which is the binary name being changed to `conduwuit` finally instead of `conduit`, but otherwise it's preferred.

**Comment by sodiboo** [comment]
> It's difficult to read it because of the very hostile tone this was written like

honestly yeah. sorry about that. i think i was just a bit pissed whilst writing it because my instance suddenly broke and there was no documentation to hold my hand. :(

**Comment by niklaskorz** [comment]
> Yes if the conduwuit module has finally landed into nixpkgs general availability then that is preferred. It does require one change though when 0.5.0 is released which is the binary name being changed to `conduwuit` finally instead of `conduit`, but otherwise it's preferred.

The module uses `lib.getExe cfg.package` for the binary name, so if you have declared `mainProgram = "conduwuit";` in this repo's flake package, then that should automatically work as `services.conduwuit.package = inputs.conduwuit.packages.${system}.default;` when combined with the module.

**Comment by niklaskorz** [comment]
> It would also be REALLY NICE to have a neat migration guide, because i notice that e.g. the database path looks very suspicious:
> 
>     * [`services.matrix-conduit` wants `/var/lib/matrix-conduit/`](https://github.com/NixOS/nixpkgs/blob/634fd46801442d760e09493a794c4f15db2d0cbb/nixos/modules/services/matrix/conduit.nix#L77)
> 
>     * [`services.conduit` wants `/var/lib/conduwuit`](https://github.com/NixOS/nixpkgs/blob/634fd46801442d760e09493a794c4f15db2d0cbb/nixos/modules/services/matrix/conduwuit.nix#L144)
> 
> 
> So just swapping out which module i use looks like it won't properly use my old database, at _least_. I could probably figure this out in an afternoon, but it feels like there ought to be a single resource that everyone can reference so it won't take them an entire afternoon.

Indeed you will have to `mv /var/lib/matrix-conduit /var/lib/conduwuit` to re-use your existing data with the new module. Besides that it should just work though.

**Comment by girlbossceo** [comment]
I don't know if there's anything that needs to be done, so I'm closing this.

**Comment by Henry-Hiles** [comment]
> I don't know if there's anything that needs to be done, so I'm closing this.

Shouldn't https://conduwuit.puppyirl.gay/deploying/nixos.html be updated to talk about `services.conduwuit`? :)

**Comment by Henry-Hiles** [comment]
Right now it says
> The flake.nix and default.nix do not currently provide a NixOS module (contributions welcome!), so [services.matrix-conduit](https://search.nixos.org/options?channel=unstable&query=services.matrix-conduit) from Nixpkgs can be used to configure conduwuit.
